src/Microsoft.Data/DataFrame.IDataView.cs

```diff
@@ -136,5 +136,86 @@ public override bool MoveNext()
+            List<DataViewSchema.Column> activeColumns = new List<DataViewSchema.Column>();
+            foreach (DataViewSchema.Column column in schema)
+            {
+                if (column.IsHidden)
+                {
+                    continue;
+                }
+                activeColumns.Add(column);
+                DataViewType type = column.Type;
+                if (type == BooleanDataViewType.Instance)
+                {
+                    columns.Add(new PrimitiveDataFrameColumn<bool>(column.Name, dataView.GetRowCount() ?? 0));
+                }
+                else if (type == NumberDataViewType.Byte)
+                {
+                    columns.Add(new PrimitiveDataFrameColumn<byte>(column.Name, dataView.GetRowCount() ?? 0));
+                }
+                else if (type == NumberDataViewType.Double)
+                {
+                    columns.Add(new PrimitiveDataFrameColumn<double>(column.Name, dataView.GetRowCount() ?? 0));
+                }
+                else if (type == NumberDataViewType.Single)
+                {
+                    columns.Add(new PrimitiveDataFrameColumn<float>(column.Name, dataView.GetRowCount() ?? 0));
+                }
+                else if (type == NumberDataViewType.Int32)
+                {
+                    columns.Add(new PrimitiveDataFrameColumn<int>(column.Name, dataView.GetRowCount() ?? 0));
+                }
+                else if (type == NumberDataViewType.Int64)
+                {
+                    columns.Add(new PrimitiveDataFrameColumn<long>(column.Name, dataView.GetRowCount() ?? 0));
+                }
+                else if (type == NumberDataViewType.SByte)
+                {
+                    columns.Add(new PrimitiveDataFrameColumn<sbyte>(column.Name, dataView.GetRowCount() ?? 0));
+                }
+                else if (type == NumberDataViewType.Int16)
+                {
+                    columns.Add(new PrimitiveDataFrameColumn<short>(column.Name, dataView.GetRowCount() ?? 0));
+                }
+                else if (type == NumberDataViewType.UInt32)
+                {
+                    columns.Add(new PrimitiveDataFrameColumn<uint>(column.Name, dataView.GetRowCount() ?? 0));
+                }
+                else if (type == NumberDataViewType.UInt64)
+                {
+                    columns.Add(new PrimitiveDataFrameColumn<ulong>(column.Name, dataView.GetRowCount() ?? 0));
+                }
+                else if (type == NumberDataViewType.UInt16)
+                {
+                    columns.Add(new PrimitiveDataFrameColumn<ushort>(column.Name, dataView.GetRowCount() ?? 0));
+                }
+                else if (type == TextDataViewType.Instance)
+                {
+                    columns.Add(new StringDataFrameColumn(column.Name, dataView.GetRowCount() ?? 0));
+                }
+                else
+                {
+                    throw new NotSupportedException(nameof(type));
```

Comment:
One issue we are going to hit immediately is that IDataView supports a "Vector" column. So in a single cell, you can store a vector of floats, for example.

We will probably need a new column type for this.

Reply:
The rest of the feedback here is easy to fix. This one will involve some work though, since a new column has to support a bunch of functions. I'm still debating the priority here with respect to shipping a preview

Reply:
Feel free to log a tracking issue for this.